L73-java-binderchannel/wireformat.md

```diff
@@ -229,7 +253,53 @@ stream transactions until we receive an acknowledgement message from the transpo
 peer. Each transport sends an acknowledgement for each 16KB of stream transaction
 received, which should avoid blocking the transport most of the time.
 
+## Stream Flow Control
+
+While transport flow control limits consumption of scarce buffer resources at
+the Android/binder layer, stream flow control limits the amount of buffering at
+the gRPC layer. The resulting back pressure lets an application adjust its rate
+of message production to match the remote reader's rate of stream consumption.
+
+### Sender Responsibilities
+Each stream sender (both client and server) must keep track of the amount of
+space remaining in its peer's receive window, decrementing this counter by the
+`bytes data` or `parcelable` size of each `rpc transaction` it sends and
+incrementing it upon receipt of a `stream receive window update`. A stream
+transaction must not be sent unless there's room for its data in the peer's
+window. Availability of peer receive window space should be exposed to the
+sending application using the language-specific stream readiness API. 
+
+The initial receive window size of all new streams is specified by a peer in its
+`setup transport transaction` and cannot be changed for the life of a transport.
+
+### Receiver Responsibilities
+A stream receiver (either client or server) must send window updates as
+its application layer requests messages and frees up space in the incoming
+transaction buffer. Window updates may be sent in their own stream transaction
+or piggybacked on message data in the opposite direction on the same stream.
+Window updates may be delayed to increase the opportunity for such piggybacking,
+or simply to combine multiple window updates into a single larger one.
+
+A receiver may increase the size of the incoming transaction buffer of an existing
+stream simply by sending a window update message advertising the additional
+capacity. A receiver can also reduce the size of this buffer by suppressing
+window updates as incoming messages are consumed by the app. However, negative
+window updates are not allowed.
+
+Receivers must detect incoming transactions that exceed their advertised receive
+window and respond by "out of band" closing the stream with canonical code
+`INTERNAL`. Receivers that reduce their incoming transaction buffer size must
+take care to account for window updates messages they sent before the reduction.
+
+### Backwards Compatibility
+Although stream flow control is a core part of the gRPC abstraction, the
+earliest drafts of this document unfortunately did not support it. For
```

Comment:
drafts -> versions? (Otherwise you might imply there's no production code which doesn't support this).

Reply:
I actually intentionally didn't say versions here because we have an int version field in the setup transaction which actually isn't changing with this update. Changed to "revisions" to make it clearer that the old wireformat has actually been widely deployed.